src/js/coblocks-lightbox.js

```diff
@@ -0,0 +1,72 @@
+/* global: Image */
+( function( $ ) {
+	'use strict';
+
+	const images = $( '.has-lightbox ul li figure img' );
+	let index;
+
+	const wrapper = $( '<div/>', { class: 'coblocks-lightbox' } );
+	const wrapperBackground = $( '<div/>', { class: 'coblocks-lightbox__background' } );
+	const modalHeading = $( '<div/>', { class: 'coblocks-lightbox__heading' } );
+	const close = $( '<button/>', { class: 'coblocks-lightbox__close' } );
+	const counter = $( '<span/>', { class: 'coblocks-lightbox__count' } );
+	const imageContainer = $( '<div/>', { class: 'coblocks-lightbox__image' } );
+	const image = $( '<img/>' );
+	const arrowLeftContainer = $( '<button/>', { class: 'coblocks-lightbox__arrow coblocks-lightbox__arrow--left' } );
+	const arrowRightContainer = $( '<button/>', { class: 'coblocks-lightbox__arrow coblocks-lightbox__arrow--right' } );
+	const arrowRight = $( '<div/>', { class: 'arrow-right' } );
+	const arrowLeft = $( '<div/>', { class: 'arrow-left' } );
+
+	modalHeading.append( counter, close );
+	imageContainer.append( image );
+	arrowLeftContainer.append( arrowLeft );
+	arrowRightContainer.append( arrowRight );
+	wrapper.append( wrapperBackground, modalHeading, imageContainer, arrowLeftContainer, arrowRightContainer );
+
+	if ( images.length > 0 ) {
+		$( 'body' ).append( wrapper );
+	}
+
+	const imagePreloader = {};
+
+	$( '.has-lightbox ul li figure img' ).click( function( e ) {
+		$.each( images, function( i, img ) {
+			imagePreloader[ 'img-' + i ] = new window.Image();
+			imagePreloader[ 'img-' + i ].src = img.attributes[ 0 ].value;
+		} );
+
+		index = Number( e.target.attributes[ 4 ].value );
+		wrapper.css( 'display', 'flex' );
+		wrapperBackground.css( 'background-image', 'url(' + imagePreloader[ 'img-' + index ].src + ')' );
+		image.attr( 'src', imagePreloader[ 'img-' + index ].src );
```

Comment:
Using the index in order to select images to load is not taking into account multiple galleries on a single page nor the group of images in a specific gallery on a page.

If we have 2 galleries on a page, each with 4 images, we will have a single index assigned to two different images. This allows only the first image with that index to be shown in the lightbox. SO if a user selected the first image on the second gallery, the first image of the first gallery would be lightboxed.

Reply:
I have adjusted the script, you can check it now, should be working good with multiple galleries @jrtashjian